br/pkg/lightning/backend/external/engine.go

```diff
@@ -402,7 +404,28 @@ func (e *Engine) ID() string {
 
 // GetKeyRange implements common.Engine.
 func (e *Engine) GetKeyRange() (startKey []byte, endKey []byte, err error) {
-	return e.startKey, e.endKey, nil
+	return DecodeKeyRangeWithAdapter(e.keyAdapter, e.startKey, e.endKey)
+}
+
+// DecodeKeyRangeWithAdapter decodes start key and end key.
+func DecodeKeyRangeWithAdapter(adapter common.KeyAdapter, start, end []byte) ([]byte, []byte, error) {
+	first, err := adapter.Decode(nil, start)
```

Comment:
in current naming convention, we have variable name pairs like start/end (end is exclusive), first/last (last is inclusive), min/max (max is inclusive). We should still use start/end in this function